### ISSUE TITLE
Blur detection methods research

### ISSUE THREAD

**Issue by ATMI** [IU-Capstone-Project-2024/A-Shot#6]
`feature/blur-detection`
@Post-Modern28

There are different methods to detect blurry image. One of the most popular is Laplacian image variance, however this approach sometimes can lead to misclassifications. As the method itself is well-known and is implemented in OpenCV library `cv2.Laplacian`, we can always safely fallback to it. That's why, I propose to study the following paper (https://paperswithcode.com/paper/depth-and-dof-cues-make-a-better-defocus-blur), which describes a new method. More papers on this topic you can find here: https://paperswithcode.com/task/defocus-blur-detection

**Comment by ATMI** [comment]
Please send any findings here!))

**Comment by ATMI** [comment]
How is going the progress?

**Comment by ATMI** [comment]
Hi, please tell me about your findings

**Comment by Post-Modern28** [comment]
Lately, several papers regarding blur detection were researched. 
  The [proposed study](https://paperswithcode.com/paper/depth-and-dof-cues-make-a-better-defocus-blur) has several problems: the manual examination of the training datased has revealed that it containes only blurred images, and moreover, with only one type of blur: defocus blur. This seriously calls into question the appropriateness of using the model for our purposes.
   I also read a [paper](https://www.scitepress.org/Papers/2020/103077/103077.pdf) which described several mathematical approaches, among which so-called HWT was showing the most outstanding and balanced results, which is also proved in [this](https://ieeexplore.ieee.org/abstract/document/1394114/) paper (actually, this research was released earlier). The method shows over 0.95 on f1, accuracy and recall scores. But the running time of the algorithm may be the problem. 
  [This](https://www.researchgate.net/publication/224116718_DCT-based_local_motion_blur_detection) paper introduced a DCT-based method, but it seems to be not so good.
  The [latter](https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13567?saml_referrer) reviewed paper uses convolutional neural network. The results beat other ML-approaches, but still it doesn't outperform HWT. Despite that, I would consider implementing this model, as it has a potential for improvements.

**Comment by ATMI** [comment]
I am closing this issue and opening a new one. I want you to focus on 2 studies:

1. https://paperswithcode.com/paper/depth-and-dof-cues-make-a-better-defocus-blur
2. https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13567?saml_referrer

Please start with the 1st one, as the researchers has published their code on GitHub: https://github.com/yuxinjin-whu/d-dffnet. Clone the repository and try to launch the model. Once it starts successfully I will give you data to test on.

If I understood correctly the 2nd option does not provide an implementation, so we will consider it later after we get some results on blur detection with the 1st one
